frameworks/Rust/warp-rust/src/main.rs

```diff
@@ -7,15 +7,16 @@ struct Message {
     message: &'static str,
 }
 
-fn main() {
-    let json = warp::path("json").map(|| {
+#[tokio::main]
+async fn main() {
+    let json = warp::path!("json").map(|| {
```

Comment:
What's the difference between the method `path` and the macro `path!` in this context? I'd like to know what rust code gets generated from this macro. Will the resulting code still instantiate and serialize the JSON response on every call?

Reply:
In short, it's a syntactic sugar for `warp::path("json").and(warp::end())`. Nothing to do with performance, just a correctness fix (previous implementation allowed paths like `/json/abc/def`.

Reply:
No, that's fine. Thanks for the explanation.